src/main/java/com/poo/demo/config/SecurityConfig.java

```diff
@@ -0,0 +1,43 @@
+package com.poo.demo.config;
+
+/**
+ * Configuração de segurança da aplicação.
+ * Define as regras de autenticação, autorização e filtros para endpoints protegidos.
+ */
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+@Configuration
+public class SecurityConfig {
+
+    private final JwtSecretProvider jwtSecretProvider;
+
+    @Autowired
+    public SecurityConfig(JwtSecretProvider jwtSecretProvider) {
+        this.jwtSecretProvider = jwtSecretProvider;
+    }
+
+    /**
+     * Configura o filtro de segurança do Spring Security.
+     * Libera o endpoint de login e protege os demais endpoints, adicionando o filtro JWT.
+     *
+     * @param http objeto de configuração do HttpSecurity
+     * @return SecurityFilterChain configurado
+     * @throws Exception em caso de erro na configuração
+     */
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        http
+            .csrf(csrf -> csrf.disable())
+            .authorizeHttpRequests(auth -> auth
+                .requestMatchers("/api/v1/auth/login").permitAll()
+                .anyRequest().authenticated()
```

Comment:
Deixar apenas pro login e cadastro estava errado, tem outras rotas do sistema que precisar de liberacao tmb
```suggestion
                .requestMatchers("/api/v1/auth/login", "/api/v1/auth/cadastrar").permitAll()
                .requestMatchers("/swagger-ui/**", "/v3/api-docs/**", "/docs/**").permitAll()
                .requestMatchers("/api/v1/health").permitAll()
                .anyRequest().authenticated()
```